File: mix/prototype/cat.rb

Cat = Class.new
def Cat.name= name
  @name = name
end
def Cat.meow
  puts "#{@name}: meow~"
end

Alice = Cat.clone
Alice.name = "Alice"
Alice.meow

Bob = Cat.clone
Bob  .name = "Bob"
Bob  .meow

Carol = Class.new(Bob)
Carol.name = "Carol"
Carol.meow

def Bob.jump
  puts "#{@name}: jump~"
end
Bob.jump

Carol.jump

# Alice.jump # no jump

File: mix/prototype/cat.js
// node cat.js
var Cat = {};
Cat.name = "";
Cat.meow = function(){ console.log(this.name + ": meow~"); };

var Alice = {__proto__: Cat};
Alice.name = "Alice";
Alice.meow();

var Bob = {__proto__: Cat};
Bob.name = "Bob";
Bob.meow();

var Carol = {__proto__: Bob};
Carol.name = "Carol";
Carol.meow();

Bob.jump = function(){ console.log(this.name + ": jump~"); };
Bob.jump();

Carol.jump();

// Alice.jump(); // no jump for Alice
